lib/rjgit.rb

```diff
@@ -190,6 +190,40 @@ def self.describe(repository, ref, options = {})
       end
       command.call
     end
+
+    # options:
+    #  * ref
+    #  * path_filter
+    #  * case_insensitive
+    def self.grep(repository, query, options={})
+      case_insensitive = options[:case_insensitive]
+      repo = RJGit.repository_type(repository)
+      walk = RevWalk.new(repo)
+      ls_tree_options = {:recursive => true, :path_filter => options[:path_filter]}
+
+      raise "A #{query.class} was passed to #{self}.grep().  Only Regexps and Strings are supported!" unless String === query || Regexp === query
+
+      if case_insensitive
+        query = case query
+        when Regexp then Regexp.new(query.source, query.options | Regexp::IGNORECASE)
+        when String then query.downcase
+        end
+      end
+
+      ls_tree(repo, nil, options.fetch(:ref, 'HEAD'), ls_tree_options).each_with_object({}) do |item, result|
+        blob = Blob.new(repo, item[:mode], item[:path], walk.lookup_blob(ObjectId.from_string(item[:id])))
+        next if blob.binary?
+
+        rows = blob.data.split("\n")
+        data = case query
+        when Regexp then rows.grep(query)
```

Comment:
Ah, didn't see the second `case` statement here. On board with @bartkamphorst's suggestion of converting string to regex. I think you can use `Regex.new` or `%r{#{string}}`. See [here](https://stackoverflow.com/questions/8652715/convert-a-string-to-regular-expression-ruby).